pkg/deployments/src/artifact.ts

```diff
@@ -37,6 +38,15 @@ function _extractArtifact(task: Task, file: string, contract?: string) {
  * @param task - The task for which to check artifact integrity.
  */
 export function checkArtifact(task: Task): void {
+  let readmeLines: string[] = [];
+
+  const filePath = path.join(task.dir(), `readme.md`);
+  const readmeFileExists = fs.existsSync(filePath) && fs.statSync(filePath).isFile();
+  if (readmeFileExists) {
+    const readmeContents = fs.readFileSync(filePath).toString();
+    readmeLines = readmeContents.split('\n');
+  }
```

Comment:
I think you could also do this inside `checkReadmeArtifactLink`, since you only need the `readmeLines` inside there, and you are already doing `if (readmeFileExists)` below. 

Instead of sending the lines, you'd send the file path, which is also available.

Reply:
That's an optimization so that I don't have to read the file (a slow operation) more than once. Usually there's only one artifact, so it doesn't make much difference, but if there are three or four (happens sometimes), it would read the file multiple times.